File: Examen Practico 1er Parcial/contar_vocales_consonantes.f
        program contar_vocales_consonantes
                implicit none
                character(len=100) :: cadena
                integer :: i, vocales, consonantes, lon
                character :: c

                vocales = 0
                consonantes = 0

                print *, 'Ingrese una cadena (max 100 caracteres):'
                read(*,'(A)') cadena

                lon = len_trim(cadena)

                do i = 1, lon
                   c = cadena(i:i)
                   if (c >= 'A' .and. c <= 'Z') then
                      c = char(iachar(c) + 32)
                   end if

                   if (c == 'a') then
                      vocales = vocales + 1
                   else if (c == 'e') then
                        vocales = vocales + 1
                   else if (c == 'i') then
                        vocales = vocales + 1
                   else if (c == 'o') then
                        vocales = vocales + 1
                   else if (c == 'u') then
                        vocales = vocales + 1
                   else if (c >= 'a' .and. c <= 'z') then
                        consonantes = consonantes + 1
                   end if
                end do


                print *, 'Cantidad de vocales: ', vocales
                print *, 'Cantidad de consonantes: ', consonantes
                print *, "Presione Enter para salir"
                read (*,*)
        end program contar_vocales_consonantes

File: Examen Practico 1er Parcial/ContarVocalesConsonantes.java
import java.util.Scanner;

public class ContarVocalesConsonantes {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String cadena;
        int vocales = 0, consonantes = 0;

        System.out.println("Ingrese una cadena:");
        cadena = sc.nextLine();

        for (int i = 0; i < cadena.length(); i++) {
            char c = Character.toLowerCase(cadena.charAt(i)); // convertir a minúscula

            if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') {
                vocales++;
            } else if (c >= 'a' && c <= 'z') {
                consonantes++;
            }
        }

        System.out.println("Cantidad de vocales: " + vocales);
        System.out.println("Cantidad de consonantes: " + consonantes);

        sc.close();
    }
}
